### DISCUSSION TITLE
What are the rate limits to the API?

### DISCUSSION THREAD

**Discussion by ghost** [mempool/mempool#752] [Q&A]
Hello,
I am building an application that will take payment on BTC, I am using mempool.space API for looking if there is any new transaction on a certain BTC address.
Now, my question is, what are the rate limits for fetching data from mempool's API per second, per minute, per hour, and per day? So that I can request for fetching data by limiting my requests according to their request limits.
Thanks in advance.

**Comment by wiz** [comment]
Our API limits aren't disclosed, basically because if you have to ask then you will hit them. For this type of application it's best if you run your own node and query it locally.

**Comment by markmhendrickson** [comment]
@wiz We're looking to integrate mempool.space's API into https://wallet.hiro.so/ but seem to be hitting limits even within limited testing in our development environment. It's not clear if the limits are really low or we're doing something wrong to provoke them. Any way we can connect to diagnose?

**Reply by wiz** [reply]
Increased API limits for enterprises is available, see mempool.space/enterprise for pricing and if you're interested we can setup a call to discuss

**Reply by markmhendrickson** [reply]
Thanks! That'd be great. I'm mark@hiro.so or happy to schedule elsewhere if there's a preferred way.

**Reply by cyaolong** [reply]
I'm from Wizz Wallet team. We have become the Enterprise silver member a few days ago. However, we can't find any API-limits there and the IP whitelist is not effective. 
Looking forward to your reply here or mails to dev@astrox.network.

**Comment by SamuelAl** [comment]
Hi @wiz , I am Sam from Sulu (www.sulu.sh). We are doing work with Lightning and L402 API monetization. Seeing some people asking about API limits, thinking would it make sense to explore adding an L402 paywall to the API and give users the option to pay-per-use or a similar monetization scheme? Would love to have this conversation and see people's interest. We can talk here or please reach out to my email samuel.alarco@sulusolutions.com :)

Please also visit our [docs](docs.sulu.sh) for more info on what we do!
